modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -742,7 +750,7 @@ def _groupby_head_tail(
         filter = transform.copy(op=FilterNode(transform, cond))
         exprs = filter._index_exprs()
         exprs.update((col, filter.ref(col)) for col in base.columns)
-        return base.copy(op=TransformNode(filter, exprs))
+        return base.copy(op=TransformNode(filter, exprs), partitions=None, index=None)
```

Comment:
why we don't copy partitions and index anymore?

Reply:
It was incorrect. The new frame will have a different index.